syntax = "proto3";

package business;

import "error/error.proto";

message Business {
    string id = 1;
    string business_name = 2;
    string email = 3;
    string website_url = 4;
    string phone_number = 5;
    string location_city = 6;
    string location_state = 7;
    string ein = 8;
    string business_description = 9;
    string business_size = 10;
}

message GetBusinessRequest {
    string business_id = 1;
}

message GetBusinessResponse {
    Business business = 1;
    repeated error.Error errors = 2;
}

message CreateBusinessRequest {
    string business_name = 1;
    string email = 2;
    string website_url = 3;
    string phone_number = 4;
    string location_city = 5;
    string location_state = 6;
    string ein = 7;
    string business_description = 8;
    string business_size = 9;
    string user_email = 10;
}

message CreateBusinessResponse {
    Business business = 1;
    repeated error.Error errors = 2;
}

service BusinessService {
    rpc GetBusiness(GetBusinessRequest) returns (GetBusinessResponse);
    rpc CreateBusiness(CreateBusinessRequest) returns (CreateBusinessResponse);
}
